core/test/mpi/distributed/vector.cpp

```diff
@@ -0,0 +1,864 @@
+    {}
+
+    void SetUp() override { ASSERT_GE(this->comm.size(), 3); }
+
+    std::shared_ptr<gko::Executor> ref;
+    gko::mpi::communicator comm;
+    std::shared_ptr<part_type> part;
+
+    md_type md;
+    md_type md_localized[3];
+};
+
+
+TYPED_TEST_SUITE(VectorCreation, gko::test::ValueLocalGlobalIndexTypes);
+
+
+TYPED_TEST(VectorCreation, CanReadGlobalMatrixData)
+{
+    using part_type = typename TestFixture::part_type;
+    using value_type = typename TestFixture::value_type;
+    auto vec = TestFixture::dist_vec_type::create(this->ref, this->comm);
+    auto rank = this->comm.rank();
+
+    vec->read_distributed(this->md, this->part.get());
+
+    GKO_ASSERT_EQUAL_DIMENSIONS(vec->get_size(), gko::dim<2>(6, 2));
+    GKO_ASSERT_EQUAL_DIMENSIONS(vec->get_local()->get_size(),
+                                gko::dim<2>(2, 2));
+    I<I<value_type>> ref_data[3] = {
+        {{0, 1}, {2, 3}},
+        {{4, 5}, {6, 7}},
+        {{8, 9}, {10, 11}},
+    };
+    GKO_ASSERT_MTX_NEAR(vec->get_local(), ref_data[rank], r<value_type>::value);
+}
+
+
+TYPED_TEST(VectorCreation, CanReadGlobalMatrixDataSomeEmpty)
+{
+    using part_type = typename TestFixture::part_type;
+    using value_type = typename TestFixture::value_type;
+    auto part = gko::share(
+        part_type::build_from_contiguous(this->ref, {this->ref, {0, 0, 6, 6}}));
+    auto vec = TestFixture::dist_vec_type::create(this->ref, this->comm);
+    auto rank = this->comm.rank();
+
+    vec->read_distributed(this->md, part.get());
+
+    GKO_ASSERT_EQUAL_DIMENSIONS(vec->get_size(), gko::dim<2>(6, 2));
+    if (rank == 1) {
+        GKO_ASSERT_EQUAL_DIMENSIONS(vec->get_local()->get_size(),
+                                    gko::dim<2>(6, 2));
+        GKO_ASSERT_MTX_NEAR(
+            vec->get_local(),
+            l({{0., 1.}, {2., 3.}, {4., 5.}, {6., 7.}, {8., 9.}, {10., 11.}}),
+            r<value_type>::value);
+    } else {
+        GKO_ASSERT_EQUAL_DIMENSIONS(vec->get_local()->get_size(),
+                                    gko::dim<2>(0, 2));
+    }
+}
+
+
+TYPED_TEST(VectorCreation, CanReadGlobalDeviceMatrixData)
+{
+    using it = typename TestFixture::global_index_type;
+    using d_md_type = typename TestFixture::d_md_type;
+    using part_type = typename TestFixture::part_type;
+    using vt = typename TestFixture::value_type;
+    d_md_type md{
+        this->ref, gko::dim<2>{6, 2},
+        gko::Array<it>{this->ref, I<it>{0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5}},
+        gko::Array<it>{this->ref, I<it>{0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1}},
+        gko::Array<vt>{this->ref, I<vt>{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}}};
+    auto part = gko::share(
+        part_type::build_from_contiguous(this->ref, {this->ref, {0, 2, 4, 6}}));
+    auto vec = TestFixture::dist_vec_type::create(this->ref, this->comm);
+    auto rank = this->comm.rank();
+    I<I<vt>> ref_data[3] = {
+        {{0, 1}, {2, 3}},
+        {{4, 5}, {6, 7}},
+        {{8, 9}, {10, 11}},
+    };
```

Comment:
you can also put it in the TestFixture as the answer of md_localized

Reply:
That gives me some segfaults, so I will keep them in each test at the moments.